libp2p/switch.nim

```diff
@@ -144,6 +145,9 @@ proc cleanupConn(s: Switch, conn: Connection) {.async, gcsafe.} =
         await s.connections[id].close()
       s.connections.del(id)
 
+    if id in s.dialedPubSubPeers:
```

Comment:
strictly, the `if` is not necessary here - it just causes an extra unnecessary lookup

Reply:
How so, `excl` doesn't fail on non existent entry, because there is a chance that there wont be one in the set.

Reply:
that's what the docs claim, that it's fine

Reply:
ok, good find